src/test/java/uk/gov/justice/probation/courtcaseservice/service/ImmutableCourtCaseServiceIntTest.java

```diff
@@ -50,6 +78,223 @@ public static AmazonSNS createAmazonSnsClient() {
+
+        HearingEntity hearingUpdateTwo = EntityHelper.aHearingEntityWithHearingId(caseId, hearingId, DEFENDANT_ID_1).withHearingEventType(HearingEventType.CONFIRMED_OR_UPDATED);
+        EntityHelper.refreshMappings(hearingUpdateTwo);
+        var savedUpdateTwo = courtCaseService.createOrUpdateHearingByHearingId(newHearingEntity.getHearingId(), hearingUpdateTwo).block();
+
+        hearingAudits = findAllAuditByHearingId(hearingId);
+        assertThat(hearingAudits).hasSize(3);
+        assertThat(hearingAudits).isSameAs(List.of(newHearingEntity, savedUpdate, savedUpdateTwo));
+    }
+  @Test
+  public void givenNewCaseWithNoDefendantOffender_createOrUpdateHearingWithOffender_shouldSafeOffender() {
+        var caseId = "fd907836-324f-42a2-8a7a-e3d956b9d1e7";
+        var hearingId = "f069bfcd-29d6-4ab0-82f4-5df1ffd47f33";
+        final var newHearingEntity = EntityHelper.aHearingEntityWithHearingId(caseId, hearingId, DEFENDANT_ID_1);
+        newHearingEntity.getHearingDefendants().get(0).setDefendant(EntityHelper.aDefendantEntity(DEFENDANT_ID_1, null));
+
+        HearingEntity savedHearing = courtCaseService.createOrUpdateHearingByHearingId(newHearingEntity.getHearingId(), newHearingEntity).block();
+        assertThat(savedHearing.withId(null)).isEqualTo(newHearingEntity.withId(null));
+
+        assertThatHearingIsNotImmutable(caseId, hearingId, DEFENDANT_ID_1);
+        var hearingAudits = findAllAuditByHearingId(hearingId);
+        assertThat(hearingAudits).hasSize(1);
+
+        // update - 1
+        HearingEntity hearingUpdate = getHearingUpdate(newHearingEntity);
+        DefendantEntity defendantUpdate = EntityHelper.aDefendantEntity(DEFENDANT_ID_1, EntityHelper.CRN).withDefendantName("Mr. Updated Name");
+        hearingUpdate.getHearingDefendants().get(0).setDefendant(defendantUpdate);
+
+        HearingEntity savedUpdate = courtCaseService.createOrUpdateHearingByHearingId(hearingUpdate.getHearingId(), hearingUpdate).block();
+        assertThat(savedUpdate.withId(null)).isEqualTo(hearingUpdate);
+        var dbHearing = assertThatHearingIsNotImmutable(caseId, hearingId, DEFENDANT_ID_1);
+        assertThat(dbHearing.getV3().getCrn()).isEqualTo(EntityHelper.CRN);
+        assertThat(offenderRepository.findByCrn(EntityHelper.CRN).get().withId(null)).isEqualTo(defendantUpdate.getOffender().withId(null));
+
+        var updatedData = assertThatHearingIsNotImmutable(caseId, hearingId, DEFENDANT_ID_1);
+        assertThat(updatedData.getV1().getHearingEventType()).isEqualTo(HearingEventType.RESULTED);
+        assertThat(updatedData.getV3().getDefendantName()).isEqualTo("Mr. Updated Name");
+    }
+
+    @NotNull
+    private HearingEntity getHearingUpdate(HearingEntity hearingEntity) {
+
+        var updatedSession = LocalDateTime.of(2040, 8, 6, 11, 30);
+        ArrayList<HearingDayEntity> updatedHearingDays = EntityHelper.getMutableList(List.of(HearingDayEntity.builder()
```

Comment:
```suggestion
        var updatedHearingDays = EntityHelper.getMutableList(List.of(HearingDayEntity.builder()
```

Reply:
getMutableList isn't returning a compatible one :( so need the type here